### PR TITLE
Serf Update

### PR DESCRIPTION
**Pull request by Kaytila**
ContinuousGFXUISelector.java
added some checks for right clicks to make the focus loosing go away.

Util.java
the random number generation was weird.

The rest is formatting, I unfortunately have a very different way of formatting.

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -13,5 +13,6 @@
 	<classpathentry kind="lib" path="lib/basicPlayer/vorbisspi1.0.2.jar"/>
 	<classpathentry kind="lib" path="lib/basicPlayer/basicplayer3.0.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/libjcsi"/>
+	<classpathentry kind="lib" path="C:/Users/Claus/Downloads/log4j-1.2.17/apache-log4j-1.2.17/log4j-1.2.17.jar"/>
```
Comment (slashman) [inline .classpath]: in case this is needed, the jar file should be added to the lib directory; we might also want to look forward to change to maven or gradle. Or may be we are just fine.

File: src/net/slashie/utils/Util.java [hunk 2]
```diff
@@ -1,113 +1,132 @@
...
+	{
 
-	public static int rand(int low, int hi){
-		//return (int)((Math.random() * (hi -low))+low);
-		//rand.setSeed(rand.nextLong());
-		int ret = (int)((rand.nextDouble() * (hi - low + 1))+low);
-		//Debug.say("low " +low+" hi "+hi+"="+ret);
+		// return (int)((Math.random() * (hi -low))+low);
+		// rand.setSeed(rand.nextLong());
+		int ret = (int) ((rand.nextDouble() * (hi - low + 1)) + low);
```
Comment (slashman) [inline src/net/slashie/utils/Util.java:18]: I don't see any change here? can you point to the actual change on Util.java?

File: src/net/slashie/serf/ui/oryxUI/ContinuousGFXUISelector.java [hunk 3]
```diff
@@ -96,195 +88,263 @@ public void keyPressed(KeyEvent e) {
...
 			}
 		};
 		final Timer gotoDirectionTimer = new Timer(200, gotoDirectionAction);
-		return new MouseAdapter() {
-			public void mousePressed(final MouseEvent e) {
+		return new MouseAdapter()
+		{
+			public void mousePressed(final MouseEvent e)
+			{
+				if (SwingUtilities.isRightMouseButton(e))
```
Comment (slashman) [inline src/net/slashie/serf/ui/oryxUI/ContinuousGFXUISelector.java:142]: Please confirm if the SwingUtilities.isRightMouseButton checks are the ones you added?

File: src/net/slashie/serf/ui/oryxUI/GFXUserInterface.java [hunk 4]
```diff
@@ -1161,6 +1166,9 @@ public void shutdown(){
 		si.cleanLayer(getItemsLayer());
 		si.commitLayer(getItemsLayer(), true);
 		
+		si.getComponentsPanel().removeAll();
```
Comment (slashman) [inline src/net/slashie/serf/ui/oryxUI/GFXUserInterface.java:1169]: Why was this needed? Memory management?

Comment (Kaytila) [inline src/net/slashie/serf/ui/oryxUI/GFXUserInterface.java:1169]: no, not memory, the Black Frame stayed otherwise, even with the game being shut down.

File: src/net/slashie/serf/ui/oryxUI/ContinuousGFXUISelector.java (hunk 3 above)
Comment (Kaytila) [inline src/net/slashie/serf/ui/oryxUI/ContinuousGFXUISelector.java:142]: yes, to get rid of the loose focus issue

File: src/net/slashie/utils/Util.java (hunk 2 above)
Comment (Kaytila) [inline src/net/slashie/utils/Util.java:18]: Good point, my notes were wrong. I reverted and its just formatting.

File: .classpath (hunk 1 above)
Comment (Kaytila) [inline .classpath]: Good point, will clean that up for the next time.

**Comment by slashman** [comment]
@Kaytila I squashed these 3 commits into a new one, in order to keep them descriptive, please rebase
